router/router_test.go

```diff
@@ -108,3 +104,129 @@ func TestRingpopRouterMakeRemoteClientCached(t *testing.T) {
 	router.GetClient("hello")
 	cf.AssertNumberOfCalls(t, "MakeRemoteClient", 1)
 }
+
+func TestRingpopRouterRemoveExistingClient(t *testing.T) {
+	rp := &mocks.Ringpop{}
+	rp.On("RegisterListener", mock.Anything).Return()
+
+	router := New(rp, nil, nil).(*router)
+
+	hostport := "127.0.0.1:3000"
+	router.clientCache[hostport] = "Some client"
+	assert.Equal(t, "Some client", router.clientCache[hostport])
```

Comment:
Is this strictly necessary? You're essentially asserting the behavior of a native Go map. Would you rather expose an API like `clientExists()`?